### PR TITLE
feat(BUILD-1512): sbom plugin

### PR DESCRIPTION
**Pull request by julien-carsique-sonarsource**
BUILD-1512: generate SBOM

Add Gradle plugin

### CODE REVIEW

**Review by tomverin — APPROVED** [review]

**Review by tomverin — CHANGES_REQUESTED** [review]
There's no bom in the build 258 https://repox.jfrog.io/ui/builds/sonar-dummy-oss/258/1649757082885/published/org.sonarsource.dummy:sonar-dummy-oss-plugin:2.5.0.258?buildRepo=artifactory-build-info

**Comment by julien-carsique-sonarsource** [comment]
```
Deploying artifact: HIDDEN-BY-CIRRUS-CI/sonarsource-public-qa/org/sonarsource/dummy/sonar-dummy-oss-plugin/2.5.0.260/sonar-dummy-oss-plugin-2.5.0.260-cyclonedx.json
```
And the file is there https://repox.jfrog.io/ui/builds/sonar-scanner-cli/2775/1649755371688/published/org.sonarsource.scanner.cli:sonar-scanner-cli:4.8.0.2775?buildRepo=artifactory-build-info

**Review by tomverin — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -81,6 +83,23 @@ subprojects {
...
+    outputs.file bomFile
+  }
+  def bomArtifact = artifacts.add('archives', bomFile.get().asFile) {
+    type 'json'
+    classifier 'cyclonedx'
+    builtBy 'cyclonedxBom'
+  }
+  publishing {
+    publications {
+      mavenJava(MavenPublication) {
```
Comment (malena-ebert-sonarsource) [inline build.gradle:98]: rename `mavenJava` to `cyclonedxBom`

**Review by malena-ebert-sonarsource — DISMISSED** [review]

File: build.gradle (hunk 1 above)
Comment (julien-carsique-sonarsource) [reply build.gradle:98]: it's the default and required by artifactoryPublish

**Pull request merged**